snapshot-labs/sx-monorepo
# feat: add read-only execution for offchain spaces

### Summary

This PR adds read only execution to offchain spaces.

Closes: https://github.com/snapshot-labs/workflow/issues/153

### How to test

1. Add this treasury to your offchain space on snapshot.org: `eth:0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f`
2. Go to your space on SX, you can create execution with this treausury.
3. Create proposal with some execution.
4. You see execution on your proposal.
5. You can download the execution.
6. You can import it in https://app.safe.global/apps/open?safe=eth:0x10a19e7ee7d7f8a52822f6817de8ea18204f2e4f&appUrl=https%3A%2F%2Fapps-portal.safe.global%2Ftx-builder
7. You can simulate the transaction and see results in Tenderly.

### Screenshots

<img width="1387" alt="image" src="https://github.com/user-attachments/assets/26229334-49ec-436a-8c42-f1a45043eb7f">

<img width="1399" alt="image" src="https://github.com/user-attachments/assets/d50e3f07-1766-4e39-868c-88460b1f1a5e">

<img width="1148" alt="image" src="https://github.com/user-attachments/assets/2d5a5395-cd23-4f18-bd87-5f5923e1913d">

<img width="1151" alt="image" src="https://github.com/user-attachments/assets/df395a03-8f94-4ac6-b516-ba83f078955b">
